From 9a209c1ca3a61b3b6408906c374bd69bc85468cf Mon Sep 17 00:00:00 2001
From: Wright Feng <wright.feng@cypress.com>
Date: Fri, 22 May 2020 03:52:01 -0500
Subject: [PATCH 113/138] non-upstream: free eventmask_msg after updating event
 mask

To avoid memory leak, we should free eventmask_msg after updating
event mask.

Memory leak is detected by kmemleak
unreferenced object 0xa94436c0 (size 64):
  comm "kworker/0:1", pid 1242, jiffies 208818 (age 20.650s)
  hex dump (first 32 bytes):
    01 03 13 15 00 00 00 00 00 00 40 00 00 04 00 00  ..........@.....
    00 00 00 80 00 00 04 00 00 00 00 00 00 00 00 00  ................
  backtrace:
    [<7f3d0f9c>] 0x7f3d0f9c
    [<7f3dd3c8>] 0x7f3dd3c8
    [<7f3d2444>] 0x7f3d2444
    [<80536630>] request_firmware_work_func+0x3c/0x64
    [<80144dd4>] process_one_work+0x1d8/0x414
    [<80145a08>] worker_thread+0x38/0x574
    [<8014a82c>] kthread+0x124/0x154
    [<801077c8>] ret_from_fork+0x14/0x2c
    [<ffffffff>] 0xffffffff

Signed-off-by: Wright Feng <wright.feng@cypress.com>

---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c
index 6f12d03..f2f6d46 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c
@@ -370,8 +370,10 @@ int brcmf_c_preinit_dcmds(struct brcmf_if *ifp)
 					       eventmask_msg, msglen);
 		if (err) {
 			brcmf_err("Set event_msgs_ext error (%d)\n", err);
+			kfree(eventmask_msg);
 			goto done;
 		}
+		kfree(eventmask_msg);
 	}
 	/* Setup default scan channel time */
 	err = brcmf_fil_cmd_int_set(ifp, BRCMF_C_SET_SCAN_CHANNEL_TIME,
-- 
1.9.1

